repos/system_upgrade/common/actors/missinggpgkeysinhibitor/libraries/missinggpgkey.py

```diff
@@ -221,23 +224,26 @@ def _get_repo_gpgkey_urls(repo):
     Returned gpgkeys are URLs with already expanded variables
     (e.g. $releasever) as gpgkey can contain list of URLs separated by comma
     or whitespaces.
+    If no gpgcheck=0 is present in the repo file, None is returned. If the
+    gpgcheck is missing or enabled and no gpgkey is present, empty list is
+    returned, which means the repo can not be checked.
     """
 
+    if not repo.additional_fields:
+        return []
+
     repo_additional = json.loads(repo.additional_fields)
 
     # TODO does the case matter here?
-    if 'gpgcheck' in repo_additional and repo_additional['gpgcheck'] in ('0', 'False', 'no'):
+    if 'gpgcheck' in repo_additional and repo_additional['gpgcheck'] in (0, '0', 'False', 'no'):
         # NOTE: https://dnf.readthedocs.io/en/latest/conf_ref.html#boolean-label
         # nothing to do with repos with enforced gpgcheck=0
-        return []
+        return None
```

Comment:
Seems to me inconsistent to return list, empty list, or None. Currently donig just fast overview, so keeping this note for later